### DISCUSSION TITLE
Shell-like string interpolator

### DISCUSSION THREAD

**Discussion by uosis** [lihaoyi/os-lib#130] [Show and tell]
We have been using a string interpolator that translates calls like

```scala
sh"echo a ${"b c"} d"
```

into

```scala
os.proc("echo", "a", "b c", "d")
```

and found it very handy. Is this something that would be worthwhile to add to os-lib itself?

**Comment by lefou** [comment]
I admit, this looks neat. But I think such a string interpolator will not bring any benefits. Or at least, there are downsides too, which may outrun the benefits.

Here are some arguments against it:

* The string context needs to be imported. This might add extra inconvenience for users who don't use wildcard imports. It may also confuse users not so familiar with os-lib. Also, when used in code snippets, where imports were often forgotten, it's hard to tell what exactly is happening.
* Dealing with whitespaces is probably harder, and can even force the user to rewrite the expression to not use the string interpolator. (see the `"b c"` arg in your example)
* The user has still to decide how the command is going to be executed (call, spawn), how to handle in- and output, and many other details. To stay concise, the next logical step would be to introduce short operators and symbolic methods, which turned out to be a pita and a huge guess games when reading the code (e.g. in `scala.sys.process` or `ammonite-ops`)
